api/src/modules/maintenance-alerts/services/maintenance-alerts.service.spec.ts

```diff
@@ -0,0 +1,296 @@
+				fromDate: '2022-02-24T16:36:06.045845',
+				page: 1,
+				size: 10,
+			});
+			expect(response.items.length).toBe(1);
+			expect(response.items[0]?.fromDate).toBe(mockMaintenanceAlert1.fromDate);
+			expect(response.page).toBe(1);
+			expect(response.size).toBe(10);
+			expect(response.total).toBe(100);
+		});
+
+		it('can filter on untilDate', async () => {
+			mockDataService.execute.mockResolvedValueOnce(
+				getDefaultMaintenanceAlertsResponse(),
+			);
+
+			const response = await maintenanceAlertsService.findAll({
+				untilDate: '2022-02-27T16:36:06.045845',
+				page: 1,
+				size: 10,
+			});
+			expect(response.items.length).toBe(1);
+			expect(response.items[0]?.untilDate).toBe(
+				mockMaintenanceAlert1.untilDate,
+			);
+			expect(response.page).toBe(1);
+			expect(response.size).toBe(10);
+			expect(response.total).toBe(100);
+		});
+
+		it('can filter on userGroups', async () => {
+			mockDataService.execute.mockResolvedValueOnce(
+				getDefaultMaintenanceAlertsResponse(),
+			);
+
+			const response = await maintenanceAlertsService.findAll(
+				{
+					page: 1,
+					size: 10,
+				},
+				{
+					userGroupIds: [
+						'0213c8d4-f459-45ef-8bbc-96268ab56d01',
+						'04150e6e-b779-4125-84e5-6ee6fc580757',
+						'0b281484-76cd-45a9-b6ce-68a0ea7f4b26',
+						'c56d95aa-e918-47ca-b102-486c9449fc4a',
+					],
+					isPersonal: true,
+				},
+			);
+			expect(response.items.length).toBe(1);
+			expect(response.items[0]?.id).toBe(mockMaintenanceAlert1.id);
+			expect(response.items[0]?.title).toBe(mockMaintenanceAlert1.title);
+			expect(response.page).toBe(1);
+			expect(response.size).toBe(10);
+			expect(response.total).toBe(100);
+		});
+	});
+
+	describe('findById', () => {
+		it('returns a single maintenance alert', async () => {
+			mockDataService.execute.mockResolvedValueOnce(
+				getDefaultMaintenanceAlertsByIdResponse(),
+			);
+			const response = await maintenanceAlertsService.findById('1');
+			expect(response.id).toBe(mockGqlMaintenanceAlert2.id);
+		});
+
+		it('throws a notfoundexception if the material request was not found', async () => {
+			const mockData: FindMaintenanceAlertsQuery = {
+				app_maintenance_alerts: [],
+				app_maintenance_alerts_aggregate: {
+					aggregate: {
+						count: 0,
+					},
+				},
+			};
+			mockDataService.execute.mockResolvedValueOnce(mockData);
+
+			try {
+				await maintenanceAlertsService.findById('unknown-id');
+			} catch (error) {
+				expect(error.response).toEqual({
+					error: 'Not Found',
+					message: "Material Request with id 'unknown-id' not found",
+					statusCode: 404,
+				});
+			}
+		});
+	});
+
+	describe('create', () => {
+		it('can create a new maintenance alert', async () => {
+			const mockData: InsertMaintenanceAlertMutation = {
+				insert_app_maintenance_alerts_one: mockGqlMaintenanceAlert1,
+			};
+			mockDataService.execute.mockResolvedValueOnce(mockData);
+
+			const response = await maintenanceAlertsService.createMaintenanceAlert({
```

Comment:
REQ: this newly created maintenance alert should be moved to mocks file. You can name it mockNewMaintenanceAlert.